WorkloadManagementSystem/Agent/JobAgent.py

```diff
@@ -469,6 +469,13 @@ def __submitJob(self, jobID, jobParams, resourceParams, optimizerParams,
     logLevel = self.am_getOption('DefaultLogLevel', 'INFO')
     defaultWrapperLocation = self.am_getOption('JobWrapperTemplate',
                                                'DIRAC/WorkloadManagementSystem/JobWrapper/JobWrapperTemplate.py')
+
+    # Add the number of requested processors to the job environment
+    if 'ExecutionEnvironment' in jobParams:
+      if isinstance(jobParams['ExecutionEnvironment'], basestring):
+        jobParams['ExecutionEnvironment'] = [jobParams['ExecutionEnvironment']]
```

Comment:
Maybe this is more precise?

```suggestion
        jobParams['ExecutionEnvironment'] = jobParams['ExecutionEnvironment'].split(';')
```
?